### DISCUSSION TITLE
Make aggregates configurable by device_class

### DISCUSSION THREAD

**Discussion by felix63** [jseidl/magic-areas#189] [Ideas / Feature Requests]
Hi,

Looks promising your solution. I am still exploring but I see that Magic Areas creates entities 'binary_sensor.area_battery_xxx' and 'sensor.area_battery_xxx'. Is there an option to turn this off other than manual excluding all battery related entities (to much work :-), for some areas this is in excess of 10 entities). Perhaps you can add this as a new feature.

**Comment by jseidl** [comment]
I believe the usefulness may vary by user, but having an option to specify which device_class you want aggregated will solve this use-case and be a cool overall enhancement of the feature. Thanks for the idea!

**Comment by felix63** [comment]
Is it an option to use the new entity_category to ignore diagnostic entities for instance?

**Reply by jseidl** [reply]
> Is it an option to use the new entity_category to ignore diagnostic entities for instance?

Possibly, will have to take a look at it

**Comment by cerebrate** [comment]
Another example of this would be "duration" type entities. While I'm probably not willing to write off their usefulness entirely, by default, aggregate sensor uptimes are probably not useful for most people...

**Reply by jseidl** [reply]
> Another example of this would be "duration" type entities. 

Yeah I can add those as "ignored by default"

**Comment by jseidl** [comment]
Good news! https://github.com/jseidl/hass-magic_areas/pull/349

Coming up next release!

**Comment by jseidl** [comment]
This is implemented in release 4.0.0 which is out!
